### PR TITLE
Mep 27: Support Basic Authentication

### PR DESCRIPTION
**Pull request by DragonDriver**
Signed-off-by: dragondriver <jiquan.long@zilliz.com>
/kind feature
See also: https://wiki.lfaidata.foundation/display/MIL/MEP+27+--+Support+Basic+Authentication

### CODE REVIEW

**Comment by DragonDriver** [comment]
/hold
Waiting for the related features to be merged in https://github.com/milvus-io/milvus

File: pymilvus/client/grpc_handler.py [hunk 1]
```diff
@@ -122,13 +126,38 @@ def _setup_grpc_channel(self):
...
+        return self._password
+
+    def reset_password(self, user, password):
+        """ reset password and then setup the grpc channel. """
+        if not isinstance(user, str) or not isinstance(password, str):
+            raise ParamError(f"invalid user {user} or password {password}")
+        if self._user == user and self._password == password:
+            return
+        self._user = user
+        self._password = password
```
Comment (XuanYang-cn) [inline pymilvus/client/grpc_handler.py:158]: Please remove these private information

File: pymilvus/client/grpc_handler.py [hunk 2]
```diff
@@ -80,6 +82,8 @@ def __init__(self, uri=config.GRPC_ADDRESS, host=None, port=None, channel=None,
         self._max_retry = kwargs.get("max_retry", 5)
 
         self._secure = kwargs.get("secure", False)
+        self._user = kwargs.get("user", None)
+        self._password = kwargs.get("password", None)
```
Comment (XuanYang-cn) [inline pymilvus/client/grpc_handler.py:86]: Please remove these private information

File: pymilvus/client/grpc_handler.py [hunk 3]
```diff
@@ -122,13 +126,38 @@ def _setup_grpc_channel(self):
                              ('grpc.enable_retries', 1),
                              ('grpc.keepalive_time_ms', 55000)]
                 )
-        self._stub = milvus_pb2_grpc.MilvusServiceStub(self._channel)
+        # avoid to add duplicate headers.
+        self._final_channel = self._channel
+        if self._user and self._password:
```
Comment (XuanYang-cn) [inline pymilvus/client/grpc_handler.py:131]: Private info

File: pymilvus/client/grpc_handler.py [hunk 4]
```diff
@@ -122,13 +126,38 @@ def _setup_grpc_channel(self):
...
         """ Server network address """
         return self._uri
 
+    @property
+    def user(self):
+        return self._user
+
+    @property
+    def password(self):
+        return self._password
```
Comment (XuanYang-cn) [inline pymilvus/client/grpc_handler.py:149]: Please remove these private information

File: pymilvus/client/grpc_handler.py (hunk 1 above)
Comment (XuanYang-cn) [inline pymilvus/client/grpc_handler.py:158]: Change the credentials in server and change the SDK client

**Review by XuanYang-cn — CHANGES_REQUESTED** [review]

**Pull request merged**